en/02_Developer_Guides/00_Model/11_Scaffolding.md

```diff
@@ -44,47 +44,84 @@ class MyDataObject extends DataObject
 > [!TIP]
 > It is typically considered a good practice to wrap your modifications in a call to [`beforeUpdateCMSFields()`](api:SilverStripe\ORM\DataObject::beforeUpdateCMSFields()) - the `updateCMSFields()` extension hook is already triggered by `parent::getCMSFields()`, so this is how you ensure any new fields are added before extensions update your fieldlist.
 
-To fully customise your form fields, start with an empty FieldList.
+To define the form fields yourself without using scaffolding, use the `tabsOnly` option in [`DataObject.scaffold_cmsfields_settings`](api:SilverStripe\ORM\DataObject->scaffold_cmsfields_settings). See [scaffolding options](#scaffolding-options) for details.
 
 ```php
 namespace App\Model;
 
+use SilverStripe\Forms\CheckboxSetField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\Forms\TextareaField;
 use SilverStripe\ORM\DataObject;
 
 class MyDataObject extends DataObject
 {
     // ...
 
+    private static array $scaffold_cmsfields_settings = [
+        'tabsOnly' => true,
+    ];
+
     public function getCMSFields()
     {
-        $fields = FieldList::create(
-            TabSet::create(
-                'Root',
-                Tab::create(
-                    'Main',
-                    CheckboxSetField::create('IsActive', 'Is active?'),
-                    TextField::create('Title'),
-                    TextareaField::create('Content')
-                        ->setRows(5)
-                )
-            )
-        );
-
-        $this->extend('updateCMSFields', $fields);
+        $this->beforeUpdateCMSFields(FieldList $fields) {
+            $fields->addFieldsToTab('Root.Main', [
+                CheckboxSetField::create('IsActive', 'Is active?'),
+                TextField::create('Title'),
+                TextareaField::create('Content')->setRows(5),
+            ]);
+        }
 
-        return $fields;
+        return parent::getCMSFields();
     }
 }
 ```
 
-> [!TIP]
-> It is good practice to invoke the `updateCMSFields()` extension hook afterward, so that extensions in modules can apply their functionality to your field list.
-
 You can also alter the fields of built-in and module `DataObject` classes by implementing `updateCMSFields()` in [your own Extension](/developer_guides/extending/extensions).
 
 > [!NOTE]
 > `FormField` scaffolding takes [`$field_labels` config](#field-labels) into account as well.
 
+## Scaffolding options
+
+`FormScaffolder` has several options that modify the way it scaffolds form fields.
+
+|option|description|
+|---|---|
+|`tabbed`|Use tabs for the scaffolded fields. All database fields and `has_one` fields will be in a "Root.Main" tab. Fields representing `has_many` and `many_many` relations will either be in "Root.Main" or in "Root.`<relationname>`" tabs.|
```

Comment:
> Fields representing `has_many` and `many_many` relations will either be in "Root.Main" or in "Root.`<relationname>`" tabs.

Shouldn't these always be in the "Root.`<relationname>`" tab? Seems like that's the point of the option?

Reply:
The tab they're in is determined by the `scaffoldFormFieldForHasMany()` and `scaffoldFormFieldForManyMany()` methods which can opt to not include the fields in their own tabs.